### PR TITLE
fix Master-slave test

### PR DESCRIPTION
**Pull request by Mixficsol**

### CODE REVIEW

**Review by chejinge — APPROVED** [review]

File: src/pika_set.cc [hunk 1]
```diff
@@ -316,12 +316,10 @@ void SUnionstoreCmd::DoThroughDB(std::shared_ptr<Slot> slot) {
...
 void SUnionstoreCmd::DoUpdateCache(std::shared_ptr<Slot> slot) {
-  std::vector<std::string> CachePrefixKeyS;
-  for (auto key : dest_key_) {
-    std::string newkey = PCacheKeyPrefixS + key;
-    CachePrefixKeyS.push_back(newkey);
-  }
   if (s_.ok()) {
+    std::vector<std::string> CachePrefixKeyS;
+    std::string newkey = PCacheKeyPrefixS + dest_key_;
+    CachePrefixKeyS.emplace_back(newkey);
```
[the remark this answers is not in the document]
Comment (Mixficsol) [reply src/pika_set.cc]: done

File: src/pika_kv.cc [hunk 2]
```diff
@@ -223,12 +223,25 @@ void DelCmd::DoThroughDB(std::shared_ptr<Slot> slot) {
...
+    std::vector<std::string> CachePrefixKey;
+    std::string PCacheKeyPrefixk;
+    std::string PCacheKeyPrefixl;
+    std::string PCacheKeyPrefixh;
+    std::string PCacheKeyPrefixs;
+    std::string PCacheKeyPrefixz;
     for (auto key : keys_) {
-      std::string newkey = PCacheKeyPrefixK + key;
-      CachePrefixKeyk.push_back(newkey);
+      PCacheKeyPrefixk = PCacheKeyPrefixK + key;
```
[the remark this answers is not in the document]
Comment (Mixficsol) [reply src/pika_kv.cc]: done

File: src/pika_set.cc [hunk 3]
```diff
@@ -411,12 +409,10 @@ void SInterstoreCmd::DoThroughDB(std::shared_ptr<Slot> slot) {
 }
 
 void SInterstoreCmd::DoUpdateCache(std::shared_ptr<Slot> slot) {
-  std::vector<std::string> CachePrefixKeyS;
-  for(auto key : dest_key_) {
-    std::string newkey = PCacheKeyPrefixS + key;
-    CachePrefixKeyS.push_back(newkey);
-  }
   if (s_.ok()) {
+    std::vector<std::string> CachePrefixKeyS;
```
[the remark this answers is not in the document]
Comment (Mixficsol) [reply src/pika_set.cc:412]: done

File: src/pika_set.cc [hunk 4]
```diff
@@ -539,6 +535,22 @@ void SMoveCmd::Do(std::shared_ptr<Slot> slot) {
...
   }
 }
+
+void SMoveCmd::DoThroughDB(std::shared_ptr<Slot> slot) {
+  Do(slot);
+}
+
+void SMoveCmd::DoUpdateCache(std::shared_ptr<Slot> slot) {
+  if (s_.ok()) {
+    std::vector<std::string> members;
```
[the remark this answers is not in the document]
Comment (Mixficsol) [reply src/pika_set.cc:543]: done

File: src/pika_zset.cc [hunk 5]
```diff
@@ -865,12 +865,10 @@ void ZInterstoreCmd::DoThroughDB(std::shared_ptr<Slot> slot) {
...
 
 void ZInterstoreCmd::DoUpdateCache(std::shared_ptr<Slot> slot) {
-  std::vector<std::string> CachePrefixKeyZ;
-  for(auto key:dest_key_){
-    std::string newkey = PCacheKeyPrefixZ + dest_key_;
-    CachePrefixKeyZ.push_back(newkey);
-  }
   if (s_.ok()) {
+    std::vector<std::string> CachePrefixKeyZ;
+    std::string newkey = PCacheKeyPrefixZ + dest_key_;
```
[the remark this answers is not in the document]
Comment (Mixficsol) [reply src/pika_zset.cc:870]: done

**Pull request merged**